src/shared/ui/DatePicker/DatePickerInput.tsx

```diff
@@ -0,0 +1,159 @@
+import React, {
+  ComponentPropsWithoutRef,
+  MutableRefObject,
+  forwardRef,
+  useEffect,
+  useState,
+} from 'react'
+import DatePicker, { Value } from 'react-multi-date-picker'
+
+import { Calendar, CalendarOutline } from '@/src/shared/assets/icons'
+import { useTranslation } from '@/src/shared/hooks'
+import { Typography } from '@/src/shared/ui'
+import clsx from 'clsx'
+
+import s from './DatePickerInput.module.scss'
+
+export type DatePickerProps = {
+  className?: string
+  data?: Date | number | string
+  defaultValue: Date | null
```

Comment:
Возможно лучше сделать это поле необязательным:
```suggestion
  defaultValue?: Date | null
```

Ибо ControlledDatePicker требует передать атрибут defaultValue, но когда я передаю он ругается типом.
Передаю я в него `new Date()` или `null`, но результат один:

![image](https://github.com/UsersIT/inctagram/assets/72882255/0cb8d47e-949a-4edc-b8b8-d3bafdd05564)
![image](https://github.com/UsersIT/inctagram/assets/72882255/fdac83d9-37ea-4339-983a-7be960e7ecaf)

Но как бы `defaultValue` то работает, то есть если передать `new Date('2022-01-01')`, то все гуд. Но вот тип ругается.

Так же я не могу поставить значение по умолчанию через `useForm`, оно буквально ничего не делает:

![image](https://github.com/UsersIT/inctagram/assets/72882255/273d0f56-7b67-4404-b0fe-cb8dda9ddb47)

Reply:
Подправила. Но что бы передать дату ее нужно преобразовать к одному из типа Values. К примеру ` defaultValue={new Date('2022-01-01').toString()}`

Reply:
type Value = Date | string | number | DateObject | null; Если передавать Date то непонятно почему ругается на типы, передаешь число или строку все нормально